src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -1,210 +1,173 @@
-  "Compresses all DLO subdirectories which have messages that have been in the
-  directory for longer than `threshold`. This will produce a timestamped .tgz
-  file in each subdirectory containing old messages."
-  [dir threshold]
-  {:pre [(or (string? dir)
-             (instance? java.io.File dir))
-         (period? threshold)]}
-  (create-metrics-for-dlo! dir)
-  (when-let [subdir-files (seq (for [subdir (remove #(already-archived? % threshold) (subdirectories dir))
-                                     :let [files (compressible-files subdir threshold)]
-                                     :when (seq files)]
-                                 [subdir files]))]
-    (time! (global-metric :compression)
-           (doseq [[subdir files] subdir-files]
-             (compress-subdir! subdir files)))))
+  (:require
+   [clojure.java.io :as io]
+   [clojure.string :as str]
+   [puppetlabs.i18n.core :refer [trs]]
+   [puppetlabs.kitchensink.core :refer [timestamp]]
+   [puppetlabs.puppetdb.nio :refer [copts copt-atomic copt-replace]]
+   [puppetlabs.puppetdb.queue :refer [cmdref->entry parse-metadata]]
+   [metrics.gauges :refer [gauge-fn]]
+   [stockpile :as stock])
+  (:import
+   [java.nio.file AtomicMoveNotSupportedException
+    FileAlreadyExistsException Files LinkOption]
+   [java.nio.file.attribute FileAttribute]))
+
+(defn- parse-cmd-filename [s]
+  (let [rx #"([0-9+])-(.*)"]
+    (when-let [[_ id qmeta] (re-matches rx s)]
+      (parse-metadata qmeta))))
+
+(defn- update-cmd-stats
+  "Updates stats to reflect the receipt of the named command."
+  [stats command size]
+  (let [addnp (fn [v n] (+ (or v 0) n))]
+    (-> stats
+        (update-in ["all" :count] inc)
+        (update-in ["all" :size] addnp size)
+        (update-in [command :count] addnp 1)
+        (update-in [command :size] addnp size))))
+
+(defn- gather-stats [dlo]
+  "Gathers statistics about the filenames directly inside the dlo
+  directory (a Path).  Does not recurse.  e.g.:
+    {\"all\" {:count 104 :size 2073951}
+     \"replace facts\" {:count 34 :size 302116}
+     \"replace catalog\" {:count 19 :size 708182}
+     ...}"
+  (with-open [path-stream (Files/newDirectoryStream dlo)]
+    (reduce (fn [stats p]
+              ;; Assume the trailing .json here and in
+              ;; entry-cmd-err-filename below.
+              (if-let [cmeta (and (.endsWith p ".json")
+                                  (parse-cmd-filename (-> p .getFileName str)))]
+                (update-cmd-stats (:command cmeta) (Files/size p))
+                stats))
+            {"all" {:count 0 :size 0}}
+            path-stream)))
+
+(defn- cmd-gauges [dlo category]
+  "Adds gauges to the dlo for the given category (e.g. \"all\"
+  \"replace command\").  The gauges will pull their values from the
+  dlo atom's :stats."
+  (let [registry (:registry @dlo)
+        stat (fn [what] (get-in (:stats @dlo) what))]
+    {:count (gauge-fn registry ["puppetlabs.puppetdb.dlo" category "count"]
```

Comment:
Got it. FWIW, in the mq_listener area of the code we call it global rather than all https://github.com/puppetlabs/puppetdb/blob/cmd-proc-refactor/src/puppetlabs/puppetdb/mq_listener.clj#L162